test/pkg/framework/http.go

```diff
@@ -156,36 +157,47 @@ func ExecuteHTTPRequestC(client *http.Client, namespace string, requestInfo HTTP
 }
 
 // ExecuteHTTPRequestWithStringResponse executes an HTTP request and returns a string response in case there is no error
-func ExecuteHTTPRequestWithStringResponse(namespace string, requestInfo HTTPRequestInfo) (string, error) {
+func ExecuteHTTPRequestWithStringResponse(namespace string, requestInfo HTTPRequestInfo) (string, string, error) {
```

Comment:
Returning content type as a function result doesn't look much flexible for me.
I think it may be better to refactor the implementation, putting the response body retrieval as a separate helper function.
That way you can adjust ExecuteHTTPRequestWithUnmarshalledResponse function to call ExecuteHTTPRequest, retrieve response body using helper function and access Content-Type header there, without exposing it here in ExecuteHTTPRequestWithStringResponse.

Reply:
Please see my latest commit